docs/best-practices/pingone/index.md

```diff
@@ -0,0 +1,248 @@
+* `pingone_environment`<a href="https://registry.terraform.io/providers/pingidentity/pingone/latest/docs/resources/environment" target="_blank">:octicons-link-external-16:</a> - Users may belong to the environment's default population.  If the environment is removed, the users within that population may be at risk.  There are platform API level controls to prevent accidental deletion of an environment where the environment's type is set to `PRODUCTION`.  `SANDBOX` environments do not have such API restriction.
+
+## Multi-team Development
+
+### Use "On-Demand" Sandbox Environments
+
+PingOne customer tenants have a "tenant-in-tenant" architecture, whereby a PingOne tenant organisation can contain many individual environments.  These individual environments can be purposed for development, test, pre-production and production purposes.  This allows for easy maintenance of multiple development and test instances.
+
+The recommended approach for multi-team development, when using a GitOps CICD promotion process, is to spin up "on-demand" development and test environments, specific to new features or to individual teams, to allow for development and integration testing that doesn't conflict with other team's development and test activities.  The Terraform provider allows administrators to use CICD automation to provision new environments as required, and remove them once the project activity no longer needs them.
+
+In a GitOps CICD promotion pipeline, configuration can be translated to Terraform config-as-code and then merged (with Pull Requests) with common test environments, where automated tests can be run.  This then allows the activities in the "on-demand" environments to be merged into a common promotion pipeline to production environments.
+
+## User Administrator Role Assignment
+
+### Use Group Role Assignments Over Terraform Managed User Role Assignments
+
+As of 24th October 2023, the PingOne platform supports assigning administrator roles groups, such that members of the group get the administrator roles assigned.
+
+Ping recommends that groups with admin role assignments are controlled by the Joiner/Mover/Leaver Identity Governance processes, separate to the Terraform CICD process that configures applications, policies, domain verification and so on.  It may be that the groups with thier role assignments are initially seeded by a Terraform.  In this case, it should still be a separate Terraform process to the process that controls platform configuration, and the user group assignments should still happen in the Joiner/Mover/Leaver Identity Governance process.
+
+Terraform can be used to assign administrator roles to individuals directly, however this is not recommended best practice except in development (or generally non-production) environments.  Ping recommends though that role assignment processes in non-production environments align as close as possible to role assignment processes in production environments.
```

Comment:
environments.  **However, Ping recommends that ** role ...

Reply:
As above comment

Reply:
as with the above, the leading "however" was superfluous